lms/security.py

```diff
@@ -143,19 +143,29 @@ def _get_lti_user(request):
         partial(bearer_token_schema.lti_user, location="form"),
         OAuthCallbackSchema(request).lti_user,
     ]
-    # Avoid checking for the LTI1.3 JWT if not there
-    if "id_token" in request.params:
-        # Don't replace all auth methods in case somehow we have a bogus token
-        # but try this one first.
-        schemas.insert(0, LTI13AuthSchema(request).lti_user)
 
     lti_user = None
-    for schema in schemas:
+    # Avoid checking for the LTI1.3 JWT if not there
+    if "id_token" in request.params:
```

Comment:
This could be extended to:

- Only check LTI authentication in the view that are going to use them (basic launch and deep linking?
- In those, decide between 1.1 vs 1.3 based on the presence of `id_token`.

Reply:
Yeah, I think it makes sense to only use the right types of authentication in the right views, rather than always trying all of them